materials/gganimate.md

```diff
@@ -0,0 +1,62 @@
+---
+layout: page
+element: notes
+title: gganimate
+language: R
+---
+
+### Installation
+
+* Some R packages are not published to CRAN
+* Can install from other places using `devtools`
+
+```
+install.packages("devtools")
+devtools::install_github("dgrtwo/gganimate")
+```
+
+### Example
+
+* Example using weather data
+* Get this data using `rnoaa`
+
+```
+install.packages('rnoaa')
+library('rnoaa')
+gnv_weather <- meteo_pull_monitors("USW00012816", date_min = "1984-01-01", date_max = "2016-12-31")
+```
+* Plot annual temperature pattern through time
+* First make a plot of the annual temperature trend using ggplot
+
+```
+library(ggplot2)
+library(gganimate)
+
+gnv_weather$year = format(gnv_weather$date, "%Y")
+gnv_weather$jday = format(gnv_weather$date, "%j")
+gnv_weather = filter(gnv_weather, tmax > 
+
+ggplot(gnv_weather, aes(x = jday, y = tmax / 10)) +
+  geom_point()
+```
+
+* Then add `frame` element to `aes()`
+
+```
+p <- ggplot(gnv_weather, aes(x = jday, y = tmax / 10, frame = year))
+  + geom_point()
+
+gganimate(p)
```

Comment:
At least for me, returns these errors and warnings: 
`sh: convert: command not found`
`Error in cmd.fun(sprintf("%s --version", convert), intern = TRUE) : `
 ` error in running command`
`In addition: Warning messages:`
`1: Removed 1 rows containing missing values (geom_point). `
`2: Removed 1 rows containing missing values (geom_point). `
`I cannot find ImageMagick with convert = 'convert'`
`Error in file(file, "rb") : cannot open the connection`
`In addition: Warning messages:`
`1: In im.convert(img.files, output = path.expand(movie.name), convert = convert,  :`
` Please install ImageMagick first or put its bin path into the system PATH variable`
`2: In file(file, "rb") :`
`  cannot open file '/var/folders/pf/rjm3plcx14q6hb9xvfqc5ht80000gn/T//RtmpESomgX/gganimate/file4812c71a7c2.gif': No such file or directory`

It looks like this requires ImageMagick?

Reply:
The `saveGIF()` loop below also throws similar errors.

Reply:
Using HTML format instead of gif works for me, for both plots, without having ImageMagick.